Document HTML Components and Clarify Page Types (#1538)
withastro/docs: src/pages/en/core-concepts/astro-pages.md

i18nReady: true
---

> **Pages** are special files that live in the `src/pages/` subdirectory. They are responsible for handling routing, data loading, and overall page layout for every HTML page in your website.

Reviewer: Question: are the files “special”? Or is it that `src/pages/` is special? I wonder if “special” here suggests you need to do something different to these files but actually putting them in `pages` was already all that was needed.

```suggestion
**Pages** are files that live in the `src/pages/` subdirectory of your Astro project. They are responsible for handling routing, data loading, and overall page layout for every HTML page in your website.
```

Author: Yep, `special` was a holdover from "special kind of Astro component", but even there it might've been unnecessary!

> ### File-based routing

Reviewer 2: ```suggestion
## File-based routing
```
Suggesting this is now ##-level, since it's no longer a ### subsection of (implied) Astro pages

> Astro leverages a routing strategy called **file-based routing.**. Each page file in your `src/pages` directory becomes an endpoint on your site based on its file path.

Reviewer: Nit: I think “file” is enough here maybe? Also do we have the convention of using a trailing slash on directories like in the opening paragraph?

```suggestion
Astro leverages a routing strategy called **file-based routing.**. Each file in your `src/pages/` directory becomes an endpoint on your site based on its file path.
```

Reviewer: I don’t think Astro likes when you put non-page files in `src/pages/` IIRC? So in practice _unless_ using the `_` prefix trick, you should have _only_ pages in `src/pages/`?

I guess similar to “special”, I’m not sure sure how I make my file a _page_ file? So wanted to focus on location rather than file type.

Reviewer 2: Chaos monkey! What if we swap positions of supported page file types and file-based routing, so that when we get to this line in routing, people already have "supported file types" in their head? Does this then become easier to write/reference?

Author: Swapping! 🙊

Write standard HTML [`<a>` elements](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a) in your component template to link between pages.

📚 Read more about [Routing in Astro](/en/core-concepts/routing/).

> ### Supported page files 

Reviewer 2: ```suggestion
## Supported page files 
```
In keeping with the page restruture, I think I'd pull this up to a ## heading, too

> Astro supports the following file types in the `src/pages` directory:

Reviewer: Not sure if this is convention but @sarah11918 can confirm!

```suggestion
Astro supports the following file types in the `src/pages/` directory:
```

Reviewer 2: Sarah can, and does, confirm!

- [`.astro`](#astro-pages)
- [`.md`](#markdown-pages)
- `.mdx` (with the [MDX Integration installed](/en/guides/markdown-content/#mdx-features))
- [`.js`/`.ts`](/en/core-concepts/astro-pages/#non-html-pages)

> - `.html` (following the same restrictions as [HTML Components](/en/core-concepts/astro-components/#html-components))

Reviewer 2: ```suggestion
- [`.html`](/en/core-concepts/astro-components/#html-components)
```
I don't think anything from the HTML components section describes anything that would limit it being a page? Yeah, the HTML component section says it's not an Astro component, but... that's kind of what this entire list is about: other things that are not astro files that are pages. I'd be tempted just to remove it. And NOW I'm going to revist that section and add a quick nod to being a full page.

## Astro Pages

Astro pages use the `.astro` file extension and support the same features as [Astro components](/en/core-concepts/astro-components/).

> ### Page HTML

Reviewer: Reading through this I’m struck that the heading “Page HTML” follows not long after we mention `.html` pages. Do you think that matters? Would renaming that heading help clarity at all or is it not an issue?

Reviewer 2: I agree. I'd remove it entirely. It made more sense when we were really ONLY talking about Astro pages here.

Looking at the right sidebar now, I'm also considering whether that layouts subheading deserves to be there, too... will report back!

Reviewer: “Required Page Content”?

Reviewer: (ah, just saw the recommendation to delete — that works too!)

> ### Page HTML

Reviewer 2: ```suggestion
```
Suggest deleting this entirely. Not needed as a heading anymore, and potentially more confusing now that we have actual HTML pages discussed on this page.

i18nReady: true
---

**Pages** are files that live in the `src/pages/` subdirectory of your Astro project. They are responsible for handling routing, data loading, and overall page layout for every page in your website.

## Supported page files 

Astro supports the following file types in the `src/pages/` directory:
- [`.astro`](#astro-pages)
- [`.md`](#markdownmdx-pages)
- `.mdx` (with the [MDX Integration installed](/en/guides/integrations-guide/mdx/#installation))
- [`.js`/`.ts`](#file-routes)
- [`.html`](#html-pages)

## File-based routing

> Astro leverages a routing strategy called **file-based routing**. Each file in your `src/pages` directory becomes an endpoint on your site based on its file path.

Reviewer 2: ```suggestion
Astro leverages a routing strategy called **file-based routing**. Each file in your `src/pages/` directory becomes an endpoint on your site based on its file path.
```

📚 Read more about [layout components](/en/core-concepts/layouts/) in Astro.

## Markdown/MDX Pages

Astro also treats any Markdown (`.md`) files inside of `src/pages/` as pages in your final website. If you have the [MDX Integration installed](/en/guides/integrations-guide/mdx/#installation), it also treats MDX (`.mdx`) files the same way. These are commonly used for text-heavy pages like blog posts and documentation.

> Page layouts are especially useful for [Markdown files.](#markdownmdx-pages) Markdown files can use the special `layout` front matter property to specify a [layout component](/en/core-concepts/layouts/) that will wrap their Markdown content in a full `<html>...</html>` page document.

Reviewer 2: ```suggestion
Page layouts are especially useful for [Markdown files](#markdownmdx-pages). Markdown files can use the special `layout` front matter property to specify a [layout component](/en/core-concepts/layouts/) that will wrap their Markdown content in a full `<html>...</html>` page document.
```

📚 Read more about [layout components](/en/core-concepts/layouts/) in Astro.

> ## Markdown Pages
> 
> Astro also treats any Markdown (`.md`) files inside of `/src/pages/` as pages in your final website. These are commonly used for text-heavy pages like blog posts and documentation.

Reviewer 2: ```suggestion
## Markdown/MDX Pages

Astro also treats any Markdown (`.md`) and (`.mdx`) files inside of `/src/pages/` as pages in your final website. These are commonly used for text-heavy pages like blog posts and documentation.
```

Author: In the table of contents, I listed `.md` and `.mdx` as separate and linked out to the `MDX` integrations page. Should I combine them and link them both to this?

Reviewer 2: I would leave the links as-is above (since this doesn't really say much!) 😅  BUT include them both here together, since they work the same way. It would be weird to have one without the other.

```

## HTML Pages

> You can place a `.html` file in the `/src/pages/` directory. This will be treated as an [HTML Component](/en/core-concepts/astro-components/#html-components) and turned into a page.

Reviewer 2: ```suggestion
Files with the `.html` file extension can be placed in the `src/pages/` and used directly as pages on your site. Note that some key Astro features are not supported in [HTML Component](/en/core-concepts/astro-components/#html-components).
```

Kind of combining how we describe Astro pages with the link to seeing HTML components for the things Astro won't do with these files.

> Files with the `.html` file extension can be placed in the `src/pages/` and used directly as pages on your site. Note that some key Astro features are not supported in [HTML Component](/en/core-concepts/astro-components/#html-components).

Reviewer 2: ```suggestion
Files with the `.html` file extension can be placed in the `src/pages/` and used directly as pages on your site. Note that some key Astro features are not supported in [HTML Components](/en/core-concepts/astro-components/#html-components).
```

```

## HTML Pages

> You can place a `.html` file in the `/src/pages/` directory. This will be treated as an [HTML Component](/en/core-concepts/astro-components/#html-components) and turned into a page.

Reviewer 2: Do we need a "provided it is a full HTML document" kind of clarification here?

Reviewer 2: Are you saying you'd like to come up with one phrase, somewhere between "nothing" and the "existing technical-sounding phrase" and use it in both places? I think that would be fine.

OR if this will still make a page even if it's not a full html page, I'm OK to just leave as is!

Author: Turns out this is just out of date. In my testing, if you don't add `<html>` in either an Astro page or an `html` page, it will get added for you, just like `DOCTYPE`.

Reviewer 2: OK, then as agreed, let's leave this as is!
